src/llm_utils.py

```diff
@@ -0,0 +1,176 @@
+    
+    for block in comment_blocks:
+        block = block.strip()
+        if not block:
+            continue
+            
+        comment_data = {
+            'path': None,
+            'line': None,
+            'body': ''
+        }
+        
+        # Validate required fields
+        has_file = False
+        has_line = False
+        has_comment = False
+        
+        for line in block.split('\n'):
+            line = line.strip()
+            if line.startswith('FILE:'):
+                comment_data['path'] = line.split('FILE:', 1)[-1].strip()
+                has_file = True
+            elif line.startswith('LINE:'):
+                try:
+                    comment_data['line'] = int(line.split('LINE:', 1)[-1].strip())
+                    has_line = True
+                except (ValueError, IndexError):
+                    continue
+            elif line.startswith('COMMENT:'):
+                comment_data['body'] += line.split('COMMENT:', 1)[-1].strip() + '\n'
+                has_comment = True
+            elif line.startswith('SUGGESTION:'):
+                suggestion = line.split('SUGGESTION:', 1)[-1].strip()
+                if suggestion and suggestion != 'N/A':
+                    comment_data['body'] += f"\n```suggestion\n{suggestion}\n```"
+            elif line:
+                comment_data['body'] += line + '\n'
+        
+        # Only add properly formatted comments
+        if has_file and has_line and has_comment:
+            comment_data['body'] = comment_data['body'].strip()
+            review_comments.append(comment_data)
+        else:
+            print(f"Skipping invalid comment block:\n{block}")
+            
+    return review_comments
+
+
+# def parse_llm_response(response: str) -> List[Dict]:
+#     """
+#     Parse the LLM response and format it into review comments
+#     Returns a list of structured comment objects
+#     """
+#     review_comments = []
+    
+#     # Split response into individual comments
+#     comment_blocks = response.split('---')
+    
+#     for block in comment_blocks:
+#         lines = [line.strip() for line in block.split('\n') if line.strip()]
+#         if not lines:
+#             continue
+            
+#         comment_data = {
+#             'body': '',
+#             'path': None,
+#             'position': 1,  # Default to first line if not specified
+#             'suggestion': None
+#         }
+        
+#         current_section = None
+        
+#         for line in lines:
+#             if line.startswith('FILE:'):
+#                 comment_data['path'] = line.split('FILE:')[1].strip()
+#             elif line.startswith('LINE:'):
+#                 try:
+#                     comment_data['position'] = int(line.split('LINE:')[1].strip())
+#                 except ValueError:
+#                     pass  # Keep default position if invalid line number
+#             elif line.startswith('COMMENT:'):
```

Comment:
The commented out code section has an issue:  `comment_data['position'] = 1` is used, it's not compatible with Github's expected format for commit review comments, because it represents the FIRST line of the file, not where changes are made. This is not correct. GitHub expects a 'position' that represents the line in the diff that the comment refers to.

```suggestion
Remove this commented out code.
```